### ISSUE TITLE
[Bug]: openjdk-17: ImageIO and JPEG

### ISSUE THREAD

**Issue by ichbindannmalwegmann** [termux/termux-packages#9563]
### Problem description

Cannot read JPEG image with ImageIO. ImageIO returns a list with readers for JPEG but the first (and only) reader is null. See the example below.

### What steps will reproduce the bug?

1. Create the Java file "TestJpeg.java":

```
import java.util.Iterator;

import javax.imageio.ImageIO;
import javax.imageio.ImageReader;

public class TestJpeg {

	public static void main( String[] args ) {
		for ( String suffix : ImageIO.getReaderFileSuffixes() ) {
			check( suffix );
		}
		try {
			System.out.println( Class.forName( "com.sun.imageio.plugins.jpeg.JPEGImageReader" ) );
		} catch ( ClassNotFoundException e ) {
			e.printStackTrace();
		}
	}

	private static void check( String suffix ) {
		Iterator<ImageReader> readers = ImageIO.getImageReadersBySuffix( suffix );
		int i = 0;
		while ( readers.hasNext() ) {
			ImageReader reader = readers.next();
			System.out.printf( "%s: %d: %s%n", suffix, i++, reader );
		}

	}

}
```

3. Compile the Java file:

```
javac TestJpeg.java 
```

4. Run the Java class:

```
java TestJpeg
```

### What is the expected behavior?

In the output you see, that for "jpg" there is one Reader but the object is null. You also see on the last line, that there is a JPEG reader "com.sun.imageio.plugins.jpeg.JPEGImageReader".

### System information

termux-info:

```
Application version:
0.118.0
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://packages-cf.termux.org/apt/termux-main/ stable main
# x11-repo (sources.list.d/x11.list)
deb https://packages.termux.org/apt/termux-x11 x11 main
Updatable packages:
All packages up to date
Android version:
11
Kernel build information:
Linux localhost 4.19.157-perf+ #1 SMP PREEMPT Mon Feb 21 18:08:36 CST 2022 aarch64 Android
Device manufacturer:
Fairphone
Device model:
FP4
```

**Comment by xtkoba** [comment]
I get this on Android 8.1.0 aarch64:

```
$ java TestJpeg
jpg: 0: com.sun.imageio.plugins.jpeg.JPEGImageReader@457e2f02
tif: 0: com.sun.imageio.plugins.tiff.TIFFImageReader@335eadca
tiff: 0: com.sun.imageio.plugins.tiff.TIFFImageReader@2b9627bc
bmp: 0: com.sun.imageio.plugins.bmp.BMPImageReader@61f8bee4
gif: 0: com.sun.imageio.plugins.gif.GIFImageReader@887af79
png: 0: com.sun.imageio.plugins.png.PNGImageReader@5bc79255
jpeg: 0: com.sun.imageio.plugins.jpeg.JPEGImageReader@23e028a9
wbmp: 0: com.sun.imageio.plugins.wbmp.WBMPImageReader@5ae63ade
class com.sun.imageio.plugins.jpeg.JPEGImageReader
```

Apparently the reader for `jpg` is not `null`. So I cannot reproduce the issue.

**Comment by ichbindannmalwegmann** [comment]
I get this:
```
$ java TestJpeg
jpg: 0: null
tif: 0: com.sun.imageio.plugins.tiff.TIFFImageReader@5f9d02cb
tiff: 0: com.sun.imageio.plugins.tiff.TIFFImageReader@5a8e6209
bmp: 0: com.sun.imageio.plugins.bmp.BMPImageReader@369f73a2
gif: 0: com.sun.imageio.plugins.gif.GIFImageReader@7d907bac
png: 0: com.sun.imageio.plugins.png.PNGImageReader@1d16f93d
wbmp: 0: com.sun.imageio.plugins.wbmp.WBMPImageReader@2b9627bc
class com.sun.imageio.plugins.jpeg.JPEGImageReader
```
Strange. It is anway strange that I get an iterator with one element, but this element is null.

**Comment by ichbindannmalwegmann** [comment]
Would be good if anyone with an other Android 11 device could test it.

**Comment by agnostic-apollo** [comment]
Same null result on my Android 11 device. Likely related to "claimed" removal in JDK 11 https://stackoverflow.com/questions/57974306/com-sun-imageio-plugins-jpeg-jpegimagereader-no-longer-part-of-jdk-11

Although, release notes don't say, other than https://www.oracle.com/java/technologies/javase/11-relnote-issues.html#JDK-8204187

Maybe try https://github.com/haraldk/TwelveMonkeys#jpeg

**Comment by xtkoba** [comment]
So this is not a bug, is it?

**Comment by ichbindannmalwegmann** [comment]
The text on https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/imageio/package-summary.html doesn't say that JPEG support has been removed, it looks like it should be available.

The text on https://www.oracle.com/java/technologies/javase/11-relnote-issues.html#JDK-8204187 says only that "No Longer Supports Images with alpha", but what I described here is that there isn't any JPEG reader, no matter if with or without alpha.

**Comment by ichbindannmalwegmann** [comment]
I mean it is strange that the iterator that should provide readers to read JPEG says that there is an element (a reader) but if you try to get the first element (reader), you just get null - that is not how it should be. Either return an iterator without any elements or return a real reader.

**Comment by agnostic-apollo** [comment]
`JPEGImageReader` is still there in `openjdk` `11` and `17` and also in `$PREFIX/opt/openjdk/jmods/java.desktop.jmod` if opened with archive viewer. Currently not sure what's the issue. 

https://github.com/openjdk/jdk11u-dev/blob/master/src/java.desktop/share/classes/com/sun/imageio/plugins/jpeg/JPEGImageReader.java

https://github.com/openjdk/jdk17u-dev/blob/master/src/java.desktop/share/classes/com/sun/imageio/plugins/jpeg/JPEGImageReader.java

Customized env didn't work either.

https://access.redhat.com/documentation/en-us/openjdk/17/html-single/using_jlink_to_customize_java_runtime_environment/index

Maybe some jmod issue.

http://openjdk.java.net/jeps/261#Packaging:-JMOD-files

https://openjdk.java.net/projects/jigsaw/quick-start

**Comment by ichbindannmalwegmann** [comment]
This doesn't help: java --add-modules ALL-SYSTEM TestJpeg

**Comment by agnostic-apollo** [comment]
> https://openjdk.java.net/projects/jigsaw/quick-start

I tried this way to create module, but it wouldn't find the `javax.imageio.plugins.jpeg` exports when using `requires`. Could have been doing something wrong. But considering `tiff` and `bmp` readers are found, not really sure what's going on. Currently, busy with other stuff to give this time.

https://github.com/openjdk/jdk17u-dev/blob/jdk-17+35/src/java.desktop/share/classes/module-info.java#L85

**Comment by ichbindannmalwegmann** [comment]
I updated all packages yesterday (includig openjdk) and now it works.

**Comment by ichbindannmalwegmann** [comment]
Can somebody confirm that?

**Comment by agnostic-apollo** [comment]
Can confirm. Likely fixed in #10511 by @xtkoba

Thanks.
